### PR TITLE
SCUMM: [RFC] Split enhancements into categories (Take 2!)

### PR DESCRIPTION
**Pull request by AndywinXp**
The purpose and description of these changes is given within the existing parent PR: https://github.com/scummvm/scummvm/pull/5102

In my own PR I have started from @eriktorbjorn great work (with his permission, of course) and I have completed it while following [this plan](https://github.com/scummvm/scummvm/pull/5102#issuecomment-1608278733).

To summarize:
* These changes convert the `enable_enhancements` boolean flag, to an `enhancements` integer (collection of bit flags);
* I have created 9 different categories of enhancements, which are organized in 4 groups in order to have an easier to use and prettier GUI checkbox interface for the user;
* I have converted all current enhancements to the new system.

This is how it looks:
image

Each checkbox has its own tooltip detailing its purpose to the user, and within `scumm/detection.h` you can find a comprehensive comment explaining the grouping rationale to future developers who want to add new enhancements.

Let me know what you think!

### CODE REVIEW

**Comment by bluegr** [comment]
Great work! I really love the end result. It's simple for users, plus adds grouping for all these enhancements. Kudos to both @eriktorbjorn and @AndywinXp :)

File: engines/scumm/metaengine.cpp [hunk 1]
```diff
@@ -249,6 +249,10 @@ bool ScummEngine::hasFeature(EngineFeature f) const {
 		(f == kSupportsQuitDialogOverride && (_useOriginalGUI || !ChainedGamesMan.empty()));
 }
 
+bool Scumm::ScummEngine::enhancementClassActive(int32 cls) {
```
Comment (sev-) [inline engines/scumm/metaengine.cpp:252]: I'd name it `enhancementOn()` or `enhancementEnabled()` for brevity, though it is a matter of taste.

Comment (AndywinXp) [reply engines/scumm/metaengine.cpp]: Thanks, done!

**Comment by sev-** [comment]
Thank you!

**Pull request merged**